validation/mock_trace_validation.go

```diff
@@ -15,30 +15,35 @@
 package validation
 
 import (
+	"fmt"
 	"reflect"
+	"sync"
 
 	"github.com/golang/protobuf/ptypes"
 
 	"google.golang.org/genproto/googleapis/devtools/cloudtrace/v2"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
 var (
 	ErrInvalidTimestamp   = status.Error(codes.InvalidArgument, "start time must be before end time")
 	ErrMalformedTimestamp = status.Error(codes.InvalidArgument, "unable to parse timestamp")
+	ErrDuplicateSpanName  = status.New(codes.AlreadyExists, "duplicate span name")
```

Comment:
These shouldn't start with Err because they are not `error`.

I also wonder if these should be exported. Can we check use these to check if an `error` return is a duplicate span name?

e.g. something like
```
if err == ErrDuplicateSpanName { ...
```

If not, maybe they shouldn't be exported at all. See https://blog.golang.org/go1.13-errors for patterns for checking for error types.

I left a note in Mohamed's PR along these lines too.

Reply:
Yep, we created an issue where we will clean all of these errors up here: https://github.com/googleinterns/cloud-operations-api-mock/issues/15